### ISSUE TITLE
redundantInit causes "initializing from a metatype value must reference 'init' explicitly"

### ISSUE THREAD

**Issue by marcelofabri** [nicklockwood/SwiftFormat#1672]
Given this code:

```swift
let a = Module.configuration.loggingEventBatchType.init(
   anyEvents: eventsToUpload
)
```

`0.53.6` removes the `.init`, causing an error:

```
error: initializing from a metatype value must reference 'init' explicitly
```

`loggingEventBatchType` is defined as `AnyLoggingEventBatch.Type`

**Comment by nicklockwood** [comment]
There's currently not really a way for SwiftFormat to detect this case since it doesn't have access to type information. The best option for now would be to disable the rule, either globally or locally using:

```swift
// swiftformat:disable:next redundantInit
let a = Module.configuration.loggingEventBatchType.init(
   anyEvents: eventsToUpload
)
```

**Comment by nicklockwood** [comment]
@marcelofabri I realized on reflection that the existing heuristic I use to avoid false positives for `redundantInit` ought to handle this case as well, so I've pushed a fix to develop that should solve it

**Comment by nicklockwood** [comment]
@marcelofabri fixed in 0.53.7
